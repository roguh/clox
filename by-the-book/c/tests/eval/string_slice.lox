print('print("howdy!"[0]);');
print("howdy!"[0]);
print('print("howdy!"[1]);');
print("howdy!"[1]);
print('print("howdy!"[2]);');
print("howdy!"[2]);
print('print("howdy!"[3]);');
print("howdy!"[3]);
print('print("howdy!"[-2]);');
print("howdy!"[-2]);
print('print("howdy!"[-1]);');
print("howdy!"[-1]);


// This is where the fun begins
print('print("howdy!"[:]);');
print("howdy!"[:]);
print('print("howdy!"[:0]);');
print("howdy!"[:0]);
print('print("howdy!"[:1]);');
print("howdy!"[:1]);
print('print("howdy!"[:-1]);');
print("howdy!"[:-1]);
print('print("howdy!"[1:-1]);');
print("howdy!"[1:-1]);
print('print("howdy!"[0:-1]);');
print("howdy!"[0:-1]);
print('print("howdy!"[1:0]);');
print("howdy!"[1:0]);
print('print("howdy!"[2:-2]);');
print("howdy!"[2:-2]);
print('print("howdy!"[0:-2]);');
print("howdy!"[0:-2]);
print('print("howdy!"[2:-1]);');
print("howdy!"[2:-1]);


print('print("howdy!"[0:]);');
print("howdy!"[0:]);
print('print("howdy!"[1:]);');
print("howdy!"[1:]);
print('print("howdy!"[-1:]);');
print("howdy!"[-1:]);





print('print(#"howdy!");');
print(#"howdy!");
