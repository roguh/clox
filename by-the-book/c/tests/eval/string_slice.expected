print("howdy!"[0]);
h
print("howdy!"[1]);
o
print("howdy!"[2]);
w
print("howdy!"[3]);
d
print("howdy!"[-2]);
y
print("howdy!"[-1]);
!
print("howdy!"[:]);
howdy!
print("howdy!"[:0]);

print("howdy!"[:1]);
h
print("howdy!"[:-1]);
howdy
print("howdy!"[1:-1]);
owdy
print("howdy!"[0:-1]);
howdy
print("howdy!"[1:0]);

print("howdy!"[2:-2]);
wd
print("howdy!"[0:-2]);
howd
print("howdy!"[2:-1]);
wdy
print("howdy!"[0:]);
howdy!
print("howdy!"[1:]);
owdy!
print("howdy!"[-1:]);

print(#"howdy!");
6
